p2p/host/autorelay/relay_finder.go

```diff
@@ -156,31 +179,18 @@ func (rf *relayFinder) background(ctx context.Context) {
 			rf.notifyMaybeConnectToRelay()
 		case <-rf.relayUpdated:
 			push = true
-		case now := <-refreshTicker.C:
-			push = rf.refreshReservations(ctx, now)
-		case now := <-backoffTicker.C:
-			rf.candidateMx.Lock()
-			for id, t := range rf.backoff {
-				if !t.Add(rf.conf.backoff).After(now) {
-					log.Debugw("removing backoff for node", "id", id)
-					delete(rf.backoff, id)
-				}
-			}
-			rf.candidateMx.Unlock()
-		case now := <-oldCandidateTicker.C:
-			var deleted bool
-			rf.candidateMx.Lock()
-			for id, cand := range rf.candidates {
-				if !cand.added.Add(rf.conf.maxCandidateAge).After(now) {
-					deleted = true
-					log.Debugw("deleting candidate due to age", "id", id)
-					delete(rf.candidates, id)
-				}
-			}
-			rf.candidateMx.Unlock()
-			if deleted {
-				rf.notifyMaybeNeedNewCandidates()
-			}
+		case <-refreshTicker.C:
+			push = rf.runScheduledWork(ctx, rf.conf.clock.Now(), scheduledWork, canCallPeerSourceChan)
```

Comment:
You could move this call to after the `select` statement instead of repeating it 5 times.